didcomm/unpack.py

```diff
@@ -0,0 +1,22 @@
+from didcomm.interfaces.did_resolver import DIDResolver
+from didcomm.interfaces.secrets_resolver import SecretsResolver
+from didcomm.types.message import Message
+from didcomm.types.mtc import MTC
+from didcomm.types.types import JSON
+from didcomm.types.unpack_result import UnpackResult, Metadata
+
+
+class Unpacker:
```

Comment:
This clashes with the naming chosen for `Packer` but I think as a consumer of a library like this, I would expect to see the `unpack` method alongside the `pack` method on the same object. Do we anticipate that a packer instance in an agent would use different `SecretsResolver`s or `DIDResolver`s than the unpacker? Or, if that question is off track, what is your reasoning for separating these operations?

Reply:
This is also something not finalized yet, but I can provide the following reasons to separate them. Feel free to provide reasoning why you think it would be better to combine them:
1) For me it looks natural to separate them due to the principle of single responsibility. 
2) Unpacker has an additional argument - `UnpackOpts`, which is not needed for `Packer`. We may need similar `PackOpts` in future for the Packer.
3) I can't provide any explicit examples from the Agent point of view where we may use different implementations of DID and Secrets resolver in Packer and Unpacker, but I would imagine they may appear.
Also please note, that I think in most of the cases the users will not pass resolvers explicitly to Packer and Unpacker, but rather register them just once for all Packers and Unpacker (that API is not there yet). We provide a possibility to pass resolvers into every instance of Packer/Unpacker to be able to override default implementations in some cases where needed.
4) `Packer` has a number of methods, not just a single `pack`. So, `unpack` there looks a bit weird in the current API design. If we change the API to have a single `pack`, then having `unpack` in the same class as well may make more sense.